### PR TITLE
[PREVIEW] Div 3369 submit dn

### PR DESCRIPTION
**Pull request by david-gimelle**
# Description

Case Formatter changed for submitting DN see Div-3393 for details

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Only automated regression test needed

**Test Configuration**:

* Hardware:
* O/S and version:
* JDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/mapper/DivorceCaseToDnCaseMapper.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    @AfterMapping
+    protected void mapDnApplicationSubmittedDate(DivorceSession divorceSession, @MappingTarget DnCaseData result) {
+        result.setDnApplicationSubmittedDate(LocalDate.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd")));
+    }
+
+    @AfterMapping
+    protected void mapConfirmPetitionDN(DivorceSession divorceSession, @MappingTarget DnCaseData result) {
+        if (
+            divorceSession.getStatementOfTruth().equalsIgnoreCase("YES")
+                || divorceSession.getStatementOfTruthNoChanges().equalsIgnoreCase("NO")
```
Comment (tchow8) [inline src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/mapper/DivorceCaseToDnCaseMapper.java]: Shouln't this still be yes?

Comment (david-gimelle) [reply src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/mapper/DivorceCaseToDnCaseMapper.java]: I have changed it

File: src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/domain/model/ccd/DnCaseData.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Data;
+
+import java.util.List;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@Data
+public class DnCaseData {
```
Comment (qzhou-hmcts) [inline src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/domain/model/ccd/DnCaseData.java]: I think on CCD has to handle all the journeys fields,  If you see CoreCaseData extends from AosCaseData, should AosCaseData extends from DnCaseData?

Comment (david-gimelle) [reply src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/domain/model/ccd/DnCaseData.java]: make sens. I have just updated AosCaseData

File: src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/domain/model/ccd/DnCaseData.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    @JsonProperty("ConfirmPetitionDN")
+    private String confirmPetitionDN;
+
+    @JsonProperty("DivorceCostsOptionDN")
+    private String divorceCostsOptionDN;
+
+    @JsonProperty("CostsDifferentDetails")
+    private String costsDifferentDetails;
+
+    @JsonProperty("statementOfTruthDN")
```
[the remark this answers is not in the document]
Comment (david-gimelle) [reply src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/domain/model/ccd/DnCaseData.java]: I can't change it, it's in the spec

**Pull request merged**